### PR TITLE
feat: enable ability to specify bootstrap token

### PR DESCRIPTION
**Pull request by markphelps**
Fixes: FLI-204

Adds the ability to specify a `bootstrap.token` and `bootstrap.expiration` for initial startup/config of auth

```yaml
  methods:
    token:
      enabled: true
      bootstrap:
        token: "flipt"
      cleanup:
        interval: 2h
        grace_period: 48h
```

```console
Version: dev
Commit: a14863b9b92d125042f63f40627f6f54c638c7e4
Build Date: 2023-02-20T20:32:38Z
Go Version: go1.19.6

2023-02-20T15:34:23-05:00	DEBUG	not a release version, disabling telemetry
2023-02-20T15:34:23-05:00	DEBUG	local state directory exists	{"path": "/Users/markphelps/Library/Application Support/flipt"}
2023-02-20T15:34:23-05:00	DEBUG	using driver	{"driver": "sqlite3"}
2023-02-20T15:34:23-05:00	DEBUG	migrations up to date
2023-02-20T15:34:23-05:00	DEBUG	store enabled	{"server": "grpc", "driver": "sqlite3"}
2023-02-20T15:34:23-05:00	INFO	access token created	{"server": "grpc", "client_token": "flipt"}
```

CleanShot 2023-02-21 at 12 32 17@2x

### CODE REVIEW

**Comment by markphelps** [comment]
@GeorgeMac one thing I dont like is that the bootstrap token never expires.

One thing we could do is make `bootstrap` a config object and give it an optional `token` and `expiration`? This would allow users to make the bootstrap process more secure by being able to use the token to create new tokens and then have the bootstrap token automatically expire. WYT?

File: internal/storage/auth/auth.go [hunk 1]
```diff
@@ -43,9 +43,10 @@ type Store interface {
 // CreateAuthenticationRequest is the argument passed when creating instances
 // of an Authentication on a target AuthenticationStore.
 type CreateAuthenticationRequest struct {
-	Method    auth.Method
-	ExpiresAt *timestamppb.Timestamp
-	Metadata  map[string]string
+	Method      auth.Method
+	ExpiresAt   *timestamppb.Timestamp
+	Metadata    map[string]string
+	ClientToken string
```
Comment (GeorgeMac) [inline internal/storage/auth/auth.go:49]: A clarifying comment here might be nice. e.g.
```suggestion
	// ClientToken is an (optional) explicit client token to be associated with the authentication.
	// When it is not supplied a random token will be generated and returned instead.
	ClientToken string
```

**Review by GeorgeMac — APPROVED** [review]
One comment suggestion; Otherwise, I like your proposal too to make it:
```yaml
authentication:
  bootstrap:
    token: "..."
    expiration: "24h" # default to infinite perhaps?
```

**Review by GeorgeMac — APPROVED** [review]
Awesome!

**Pull request merged**